contracts/risk/CoverageDataProvider.sol

```diff
@@ -0,0 +1,406 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.6;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+
+import "../interfaces/risk/ICoverageDataProvider.sol";
+import "../interfaces/utils/IRegistry.sol";
+import "../interfaces/utils/IVault.sol";
+import "../interfaces/ISOLACE.sol";
+import "../interfaces/utils/Aave/IAavePriceOracle.sol";
+import "../interfaces/utils/Sushiswap/ISushiswapLPToken.sol";
+import "../utils/Governable.sol";
+
+/**
+ * @title  CoverageDataProvider
+ * @author solace.fi
+ * @notice Calculates the maximum amount of cover that `Solace` protocol can sell as a coverage. 
+*/
+contract CoverageDataProvider is ICoverageDataProvider, Governable {
```

Comment:
How can we refactor this contract for the cross-chain?

Reply:
Some initial and incomplete thoughts (this is a hard question about cross-chain architecture imo, more so than this smart contract itself):

The Ethereum mainnet dependencies that I can identify in this contract are
- AaveOracle.sol
- SOLACE-USDC Sushiswap LP pool
- Vault
- Underwriting pools

Questions to answer:
- Will each chain’s Solace app read balances from vaults, underwriting pools and SOLACE liquidity pools from just the chain it is on, or from all chains that they are deployed on?
- Will we deploy a new vault, underwriting pool, SOLACE liquidity pools for each chain? 
- Can we make a cross-chain call, e.g. if we call the view function CoverageDataProvider.maxCover() on Aurora, can this query Ethereum mainnet smart contract balances?
- What is a secure, reliable and swift solution for achieving cross-chain calls?

Reply:
Vault was deprecated for UWPs. Aave and Sushiswap don't exist on all chains (eg Aurora). We might switch to another dex or create another sushi pool. I'm not sure the best way to add the necessary flexibility

Reply:
Ok, I think the best reliable solution here is to have CoverageDataProvider to store individual UWP balances as variables for now, including it's own chain, that governance (define new underwritingAccountant role) can update, we could automate through a bot, before streamlining cross-chain calls. So @ssozuer or @kyzooghost just make like a simple mapping/ledger and a sum function that risk manager will read to get total UWP